packages/react-mutation-mapper/src/util/CivicUtils.ts

```diff
@@ -0,0 +1,203 @@
+                        promises.push(addCivicVariant(civicVariants,
+                                                      geneEntry.variants[proteinChange],
+                                                      proteinChange,
+                                                      geneSymbol,
+                                                      geneEntry.id));
+                    }
+                }
+            }
+        }
+    } else {
+        for (let geneName in civicGenes) {
+            let geneEntry = civicGenes[geneName];
+            let geneVariants = geneEntry.variants;
+            if (!_.isEmpty(geneVariants)) {
+                for (let variantName in geneVariants) {
+                    // Only retrieve CNA variants
+                    if (variantName == 'AMPLIFICATION' || variantName == 'DELETION') {
+                        promises.push(addCivicVariant(civicVariants,
+                                                      geneVariants[variantName],
+                                                      variantName,
+                                                      geneName,
+                                                      geneEntry.id));
+                    }
+                }
+            }
+        }
+    }
+
+    // We're explicitly waiting for all promises to finish (done or fail).
+    // We are wrapping them in another promise separately, to make sure we also 
+    // wait in case one of the promises fails and the other is still busy.
+    return Promise.all(promises).then(function() {
+        return civicVariants;
+    });
+}
+
+/**
+ * Build a Civic Entry with the data given.
+ */
+export function buildCivicEntry(geneEntry: ICivicGeneData, geneVariants: {[name: string]: ICivicVariantData}) {
+    return {
+        name: geneEntry.name,
+        description: geneEntry.description,
+        url: geneEntry.url,
+        variants: geneVariants
+    };
+}
+
+/**
+ * Returns an ICivicEntry if the civicGenes and civicVariants have information about the gene and the mutation (variant) specified. Otherwise it returns null.
+ */
+export function getCivicEntry(mutation: Mutation, civicGenes:ICivicGene, civicVariants:ICivicVariant): ICivicEntry | null
+{
+    let geneSymbol: string = mutation.gene? mutation.gene.hugoGeneSymbol : "";
+    let civicEntry = null;
+    //Only search for matching Civic variants if the gene mutation exists in the Civic API
+    if (civicGenes[geneSymbol] && civicVariants[geneSymbol] && civicVariants[geneSymbol][mutation.proteinChange]) {
+        let geneVariants: {[name: string]: ICivicVariantData} = {[mutation.proteinChange]: civicVariants[geneSymbol][mutation.proteinChange]};
+        let geneEntry: ICivicGeneData = civicGenes[geneSymbol];
+        civicEntry = buildCivicEntry(geneEntry, geneVariants);
+    }
+
+    return civicEntry;
+}
+
+export function getCivicStatus(civicGenesStatus:"pending" | "error" | "complete",
```

Comment:
there is function which aggregates the status of a set of mobxpromises.  looking for it.  i think it could be used here.

Reply:
getMobxPromiseGroupStatus